### PR TITLE
fix: broadcasting empty stream

### LINKED ISSUE #1674: bug: Broadcast message errors when sending empty stream
## Description

Apart from this, we also want to fix broadcast message rpc. It errors when trying to send an empty stream (using `ClientExchange(Send/Recv)Exec`s:

**`abc.csv`**:

```csv
id,name
```

**`glaredb local`:**

```
> create external table abc from local (location 'abc.csv');
Table created
> select * from abc;
Error: Execution error: failed to stream broadcast exchange: Status { code: Unknown, message: "Remote session error: stream missing first IPC batch", metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Wed, 30 Aug 2023 17:22:15 GMT", "content-length": "0"} }, source: None }
```

### PR DESCRIPTION
**Pull request by adhish20**
Fixes #1674

### CODE REVIEW

**Comment by scsmithr** [comment]
Is there a test case that would get this fail?

Running these queries on main and not seeing errors:

```
> create table empty_test as select 1 where false;
Table created
> select * from empty_test;
┌┐
└┘
> insert into empty_test values (1);
Inserted 1 row
> select * from empty_test;
┌──────────┐
│ Int64(1) │
│ ──       │
│ Int64    │
╞══════════╡
│ 1        │
└──────────┘
```

**Comment by adhish20** [comment]
> Is there a test case that would get this fail?
> 
> Running these queries on main and not seeing errors:
> 
> ```
> > create table empty_test as select 1 where false;
> Table created
> > select * from empty_test;
> ┌┐
> └┘
> > insert into empty_test values (1);
> Inserted 1 row
> > select * from empty_test;
> ┌──────────┐
> │ Int64(1) │
> │ ──       │
> │ Int64    │
> ╞══════════╡
> │ 1        │
> └──────────┘
> ```

running with remote server running should error on main

**Comment by scsmithr** [comment]
> > Is there a test case that would get this fail?
> > Running these queries on main and not seeing errors:
> > ```
> > > create table empty_test as select 1 where false;
> > Table created
> > > select * from empty_test;
> > ┌┐
> > └┘
> > > insert into empty_test values (1);
> > Inserted 1 row
> > > select * from empty_test;
> > ┌──────────┐
> > │ Int64(1) │
> > │ ──       │
> > │ Int64    │
> > ╞══════════╡
> > │ 1        │
> > └──────────┘
> > ```
> 
> running with remote server running should error on main

^ That was connected to a cloud deployment.

Running locally:

```
GlareDB (v0.4.0)
Type \help for help.
Using in-memory catalog
> \open http://localhost:6542
Connected to Cloud deployment: unknown
> create table empty_test as select 1 where false;
Table created
> select * from empty_test;
┌┐
└┘
> insert into empty_test values (1);
Inserted 1 row
> select * from empty_test;
┌──────────┐
│ Int64(1) │
│ ──       │
│ Int64    │
╞══════════╡
│ 1        │
└──────────┘
```

**Comment by adhish20** [comment]
try external table

**Comment by scsmithr** [comment]
Yeah I got it to fail:

```
> create external table abc from local (location 'abc.csv');
Table created
> select * from abc;
Error: Execution error: failed to stream broadcast exchange: Status { code: Unknown, message: "Remote session error: stream missing first IPC batch", metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Thu, 31 Aug 2023 17:49:18 GMT", "content-length": "0"} }, source: None }
```

Kinda strange that this doesn't already return an empty batch.

File: crates/sqlexec/src/planner/physical_plan/client_send.rs [hunk 1]
```diff
@@ -151,17 +157,26 @@ struct ClientExchangeSendStream {
     /// record batch stream), and producing a stream of ipc encoded data, we
     /// don't really have a decent way forwarding errors through the stream.
     result: Arc<Mutex<ClientExchangeSendResult>>,
+
+    // needed to create an empty reocrd batch in case of a empty stream
+    schema: Arc<Schema>,
```
Comment (scsmithr) [inline crates/sqlexec/src/planner/physical_plan/client_send.rs:162]: Do we need this? Could we do `self.stream.schema()` in `poll_next`?

Comment (adhish20) [reply crates/sqlexec/src/planner/physical_plan/client_send.rs:162]: yeah, we could do that. fixing

**Pull request merged**